apps/sms/js/activity_handler.js

```diff
@@ -229,25 +229,24 @@ var ActivityHandler = {
     //Validate if message still exists before opening message thread
     //See issue https://bugzilla.mozilla.org/show_bug.cgi?id=837029
     if (!message) {
-      return;
+      return Promise.reject(new Error('Message is not defined!'));
     }
 
-    MessageManager.getMessage(message.id).then((message) => {
+    return MessageManager.getMessage(message.id).then((message) => {
       if (!Threads.has(message.threadId)) {
         Threads.registerMessage(message);
       }
 
       if (Compose.isEmpty()) {
-        ActivityHandler.toView(message);
-        return;
+        return ActivityHandler.toView(message);
       }
 
-      Utils.confirm('discard-new-message').then(() => {
+      return Utils.confirm('discard-new-message').then(() => {
```

Comment:
I'm not sure of this "return", or maybe you should catch the possible rejection from "confirm" ?